src/wasi/WASI.h

```diff
@@ -46,6 +46,7 @@ class WASI {
     F(fd_write, I32I32I32I32_RI32)                         \
     F(fd_tell, I32I32_RI32)                                \
     F(fd_read, I32I32I32I32_RI32)                          \
+    F(fd_pread, I32I32I32I64I32_RI32)                      \
```

Comment:
What is the difference between pread and read?

Reply:
The only difference is that pread does not modify or use the file descriptor's offset when reading.